### PR TITLE
Run extra tests to check file header failure on backup file download

### PR DESCRIPTION
**Pull request by Itxaka**
We were not checking if the tests were passing in case of the file headers not being correct. This adds the same tests as when the backup success but passing empty headers as to check for proper download process.

### CODE REVIEW

**Comment by skazi0** [comment]
Would it be possible to check if the default filename was used as a fallback?

**Comment by Itxaka** [comment]
Yup, it is possible with some regex magic against the downloadBackupMock. Will add tomorrow!

File: assets/app/features/upgrade/controllers/upgrade-backup.controller.spec.js [hunk 1]
```diff
@@ -114,7 +114,51 @@ describe('Upgrade Flow - Backup Controller', function() {
...
+                        });
+
+                        // Run the backup get function
+                        controller.backup.download(42);
+                        $rootScope.$digest();
+                    });
 
+                    it('click() method should have been triggered to download the backup', function () {
+                        expect($document[0].createElement).toHaveBeenCalledWith('a');
+                        expect(downloadBackupMock.click).toHaveBeenCalled();
```
Comment (santiph) [inline assets/app/features/upgrade/controllers/upgrade-backup.controller.spec.js:147]: Could you also add `expect(downloadBackupMock.click.calls.count()).toBe(1);` so we assure only once this is being called?

Comment (santiph) [reply assets/app/features/upgrade/controllers/upgrade-backup.controller.spec.js:147]: or `assert.isTrue(downloadBackupMock.click.calledOnce);`

**Comment by santiph** [comment]
Minor comment. Please address it and consider this PR approved

File: assets/app/features/upgrade/controllers/upgrade-backup.controller.spec.js (hunk 1 above)
Comment (Itxaka) [reply assets/app/features/upgrade/controllers/upgrade-backup.controller.spec.js:147]: Sounds good!, will do

**Comment by Itxaka** [comment]
Updated to fix the comments and add the extra checks for the filename

**Review by santiph — APPROVED** [review]
Looks good to me. Approved

**Pull request merged**